From fa777598f0ad5bc8193c0a3d450c5e25c5dddd74 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Mon, 26 Feb 2018 17:47:04 +0200
Subject: [PATCH 0540/5242] MLK-10629-1 dts: imx7d-sdb: add the integrated
 Broadcom WiFi chip support

commit  74c8ef431f8dcd4e17bcce3f0356a30e3df66b97 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add Broadcom WiFi chip BCM4339 on board support.
Need adds a bcmdhd_wlan@0 node for WiFi driver to probe properly.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
Signed-off-by: Haibo Chen <haibo.chen@freescale.com>

Upstream already added usdhc2 with a few extra args, kept their
definition.

Upstream added regulator with same gpio but different voltage? Kept
both definitions.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-sdb.dts |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index a5411f5..f86c072 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -137,6 +137,17 @@
 		enable-active-high;
 	};
 
+	wlreg_on: regulator-wl {
+		compatible = "regulator-fixed";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		regulator-name = "wlreg_on";
+		gpio = <&gpio4 21 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
 	panel {
 		compatible = "innolux,at043tn24";
 		pinctrl-0 = <&pinctrl_backlight>;
@@ -158,6 +169,11 @@
 		status = "okay";
 	};
 
+	bcmdhd_wlan_0: bcmdhd_wlan@0 {
+		compatible = "android,bcmdhd_wlan";
+		wlreg_on-supply = <&wlreg_on>;
+	};
+
 	pxp_v4l2_out {
 		compatible = "fsl,imx7d-pxp-v4l2", "fsl,imx6sx-pxp-v4l2", "fsl,imx6sl-pxp-v4l2";
 		status = "okay";
-- 
1.7.9.5

